akka-typed/src/main/java/akka/typed/javadsl/Actor.java

```diff
@@ -371,4 +371,74 @@ else if (Behavior.isAlive(behv))
     return new Deferred<T>(producer);
   }
 
+  /**
+   * Factory for creating a <code>MutableBehavior</code> that typically holds mutable state as
+   * instance variables in the concrete <code>MutableBehavior</code> implementation class.
+   *
+   * Creation of the behavior instance is deferred, i.e. it is created via the <code>producer</code>
+   * function. The reason for the deferred creation is to avoid sharing the same instance in
+   * multiple actors, and to create a new instance when the actor is restarted.
+   *
+   * @param producer
+   *          behavior factory that takes the child actor’s context as argument
+   * @return the deferred behavior
+   */
+  static public <T> Behavior<T> mutable(akka.japi.function.Function<ActorContext<T>, MutableBehavior<T>> producer) {
+    return deferred(ctx -> producer.apply(ctx));
+  }
```

Comment:
What about having an override of this method with a `Supplier` which would be useful for mutable behaviour constructors that do not use context?